### PR TITLE
Start krnlmon thread once P4runtime server is started

### PR DESCRIPTION
**Pull request by nupuruttarwar**
Signed-off-by: Nupur Uttarwar <nupur.uttarwar@intel.com>

### CODE REVIEW

File: stratum/hal/lib/tdi/dpdk/dpdk_hal.cc [hunk 1]
```diff
@@ -19,6 +19,7 @@
 #include "stratum/lib/constants.h"
 #include "stratum/lib/macros.h"
 #include "stratum/lib/utils.h"
+#include "krnlmon_main.h"
```
Comment (ffoulkes) [inline stratum/hal/lib/tdi/dpdk/dpdk_hal.cc:24]: This creates an unconditional dependency on krnlmon, meaning that we can no longer build stratum without it.

I think the krnlmon code should be conditionalized (e.g. `#ifdef WITH_KRNLMON`). You'd need to add:
```
if(WITH_KRNLMON)
    target_compile_definitions(stratum_main_o PRIVATE WITH_KRNLMON)
endif()
```
to stratum/CMakeLists.txt.

Incidentally, you can remove
```
target_compile_definitions(stratum_main_o PRIVATE SDE_9_5_0)
```
while you're at it. This definition is obsolete.

Comment (nupuruttarwar) [reply stratum/hal/lib/tdi/dpdk/dpdk_hal.cc:24]: Done

Comment (ffoulkes) [reply stratum/hal/lib/tdi/dpdk/dpdk_hal.cc:24]: My recommendation was that you not use this module at all if krnlmon isn't enabled. You've opted for a messier, more difficult to maintain solution, over a simple one.

**Review by ffoulkes — APPROVED** [review]
I think you're going down the wrong road here, but I have no desire to spend my PTO trying to convince you otherwise.

**Comment by nupuruttarwar** [comment]
> I think you're going down the wrong road here, but I have no desire to spend my PTO trying to convince you otherwise.

This will be reworked after this drop for cleaner solution

**Pull request merged**